packages/hardhat/contracts/StreamFactory.sol

```diff
@@ -47,6 +55,9 @@ contract StreamFactory is AccessControl, Ownable {
         bool _startsFull,
         IERC20 _gtc
     ) public isPermittedFactoryManager returns (address streamAddress) {
+        User storage user = users[_toAddress];
+        require(user.hasStream == false, "User already has a stream!");
```

Comment:
I think that maybe you can check this without the new users mapping, just checking userStreams[_toAddress] != address(0).

Reply:
how does that check for an existing stream for that user, that is the zero address?

Reply:
The mapping to address is initialized with address(0), so you can check if the userStream for a user address was not set comparing the value with address(0)